pkg/keyspace/tso_keyspace_group.go

```diff
@@ -143,6 +152,30 @@ func (m *GroupManager) Close() {
 	m.wg.Wait()
 }
 
+func (m *GroupManager) allocNodesForDefaultKeyspaceGroup() {
+	defer logutil.LogPanic()
+	defer m.wg.Done()
+	ticker := time.NewTicker(allocNodesForDefaultKeyspaceGroupInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-m.ctx.Done():
+			return
+		case <-ticker.C:
+		}
+		kg, err := m.GetKeyspaceGroupByID(utils.DefaultKeyspaceGroupID)
+		replica := m.GetNodesNum()
+		if err == nil && kg != nil && len(kg.Members) >= replica {
```

Comment:
from this line, if my understanding is correct, we want to assign a replica/member of default keyspace group to evernt tso node, but this isn't true. Currently tso service use this logic, because the kg.Members is empty. I remember we can specify the desired replica/member count (default value is 2 if it isn't specified) of a keyspace group when creating a keyspace group, so we should use that desired replica/member count instead of GetNodesNum() here.

Reply:
https://github.com/tikv/pd/pull/6342#discussion_r1172046709